File: examples/go/hello.go
#!/usr/bin/env -S shebangsy go

/*
Minimal Go hello-world: prints a fixed one-line greeting to stdout.

Usage:
	./examples/go/hello.go

Expected:
	hello from go
*/

package main

import "fmt"

// main prints a one-line hello from the Go shebangsy example.
func main() {
	fmt.Println("hello from go")
}

File: examples/cpp/hello.cpp
#!/usr/bin/env -S shebangsy cpp

// Minimal C++ hello-world: prints a fixed greeting to stdout.
//
// Usage:
//   ./examples/cpp/hello.cpp
//
// Expected:
//   Hello, C++!

#include <iostream>

using namespace std;

// Prints a short hello message to stdout.
int main() {
  auto message = "Hello, C++!";
  cout << message << endl;
  return 0;
}
